### PR TITLE
Fix bug to get column names correctly in the ORC scanner (#2939)

### LINKED ISSUE #2939: No data returned from query of iceberg external table with orc format
### Steps to reproduce the behavior (Required)
1. create iceberg table and insert data:
```
create table hive_prod.iceberg_db.iceberg_orc_snappy(
 col_boolean boolean,
 col_int int,
 col_long long,
 col_float float,
 col_double double,
 col_decimal decimal(38,18),
 col_date date,
 col_timestamp timestamp,
 col_string string,
 col_binary binary
)
USING iceberg 
TBLPROPERTIES('write.format.default'='orc','write.parquet.compression-codec'='snappy');
insert into hive_prod.iceberg_db.iceberg_orc_snappy select true,1,1.01,1.001,1.0001,10000000000.0000000001,cast('2020-01-01' as date),cast('2020-01-01 00:00:01' as timestamp),'Top 10 Unsolved Mysteries of Paleontological Dinosaurs, Did You Know?',cast('1110001010101011001001' as binary);
```
```
spark-sql> select * from hive_prod.iceberg_db.iceberg_orc_snappy;
22/01/19 15:53:05 WARN [main] HiveConf: HiveConf of name hive.metastore.delta.compatible.mode.enabled does not exist
true	1	1	1.001	1.0001	10000000000.000000000100000000	2020-01-01	2020-01-01 00:00:01	Top 10 Unsolved Mysteries of Paleontological Dinosaurs, Did You Known	1110001010101011001001
Time taken: 4.75 seconds, Fetched 1 row(s)
```
2. create iceberg external table and select:
```
mysql>  CREATE EXTERNAL TABLE ex_iceberg_tbl (
    ->   col_boolean boolean,
    ->   col_int int,
    ->   col_string varchar(1000)
    ->  ) ENGINE=ICEBERG
    ->  PROPERTIES ( "resource" = "iceberg_emr_tn", "database" = "iceberg_db", "table" = "iceberg_orc_snappy" );
Query OK, 0 rows affected (0.03 sec)

mysql> select * from ex_iceberg_tbl;
Empty set (0.09 sec)
```

### Expected behavior (Required)
1 row returned

### Real behavior (Required)
no data returned

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
mysql> select current_version();
+------------------------+
| current_version()      |
+------------------------+
| QA_TEST_MASTER a0b5c59 |
+------------------------+
1 row in set (0.00 sec)
```

### PR DESCRIPTION
**Pull request by Ielihs**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #2939

## Problem Summary(Required) ：

Fix bug to get column names correctly in the ORC scanner.

### CODE REVIEW

**Review by Astralidea — DISMISSED** [review]

File: be/src/exec/vectorized/orc_scanner_adapter.cpp [hunk 1]
```diff
@@ -1046,7 +1046,7 @@ void OrcScannerAdapter::build_column_name_set(std::unordered_set<std::string>* n
                                               const std::vector<std::string>* hive_column_names,
                                               const orc::Type& root_type) {
     name_set->clear();
-    if (hive_column_names != nullptr) {
+    if (hive_column_names != nullptr and hive_column_names->size() > 0) {
```
Comment (caneGuy) [inline be/src/exec/vectorized/orc_scanner_adapter.cpp:1049]: nit: and -> && in case of unified style?

**Review by caneGuy — DISMISSED** [review]

**Review by DorianZheng — APPROVED** [review]
LGTM

**Review by dirtysalt — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by DorianZheng** [comment]
run starrocks_be_unittest

**Comment by Astralidea** [comment]
run starrocks_be_unittest

**Comment by wangruin** [comment]
run starrocks_be_unittest

**Comment by wangruin** [comment]
run starrocks_be_unittest

**Pull request merged**